src/data/repositories/DataSetPeriodDateD2Repository.ts

```diff
@@ -0,0 +1,221 @@
+                  return `${period.year}=${period.startDate}-${period.endDate}`;
+              })
+            : [];
+
+        return {
+            periodDate: {
+                attribute: { id: attributes.periodDates.id },
+                value: periods.join(","),
+            },
+        };
+    }
+
+    private getAllDataSets(
+        initialPage: number,
+        dataSets: DataSetPeriodDate[],
+        config: D2Config
+    ): FutureData<DataSetPeriodDate[]> {
+        return this.getDataSets(initialPage, config).flatMap(response => {
+            const entities = response.objects.map(d2Object => {
+                return {
+                    id: d2Object.id,
+                    name: d2Object.name,
+                    outcomeDate: this.buildPeriodDateFromAttributes(
+                        d2Object,
+                        config.attributes,
+                        "outcomeDates"
+                    ),
+                    outputDate: this.buildPeriodDateFromAttributes(
+                        d2Object,
+                        config.attributes,
+                        "outputDates"
+                    ),
+                    periodDate: this.buildPeriodDateFromAttributes(
+                        d2Object,
+                        config.attributes,
+                        "periodDates"
+                    ),
+                };
+            });
+            const newDataSets = [...dataSets, ...entities];
+            if (response.pager.page >= response.pager.pageCount) {
+                return Future.success(newDataSets);
+            } else {
+                return this.getAllDataSets(initialPage + 1, newDataSets, config);
+            }
+        });
+    }
+
+    private getDataSets(page: number, config: D2Config) {
+        return apiToFuture(
+            this.api.models.dataSets.get({
+                fields: {
+                    id: true,
+                    name: true,
+                    attributeValues: { attribute: { id: true }, value: true },
+                },
+                filter: {
+                    // id: { in: ["S8Z1Ed9KXf2", "bUagyhGy9rA", "u7LxjTyGV54"] },
+                    "attributeValues.attribute.id": { eq: config.attributes.inputDates.id },
+                },
+                page: page,
+                pageSize: 200,
+            })
+        ).map(response => response);
+    }
+
+    private buildPeriodDateFromAttributes(
+        d2DataSet: D2PeriodDataSet,
+        attributes: D2Config["attributes"],
+        propertyName: keyof D2Config["attributes"]
+    ): PeriodDate {
+        const inputDate = d2DataSet.attributeValues.find(
+            attribute => attribute.attribute.id === attributes.inputDates.id
+        );
+        const periodDate = d2DataSet.attributeValues.find(
+            attribute => attribute.attribute.id === attributes[propertyName].id
+        );
+
+        const [startDate, endDate] = inputDate?.value.split("-") ?? ["", ""];
```

Comment:
Not that I am seeing a lot of split("-"), if they are the "same" split (and join("-") for the inverse direction, it would be convenient to have centralized abstracted functions for these split/join. So we know we are acting on the same concept (imagine that at some point we need to change - with a _, only one point of code should change)